tests/embed/test_omni.py

```diff
@@ -189,3 +190,26 @@ def run(diag_aug):
 
         run(diag_aug=True)
         run(diag_aug=False)
+
+    def test_omni_embed_sparse(self):
+        def compute_bar(arr):
+            n = arr.shape[0] // 2
+            return (arr[:n] + arr[n:]) / 2
+
+        def run(diag_aug):
+            X, A1, A2 = generate_data(1000, seed=2)
+            Abar = (A1 + A2) / 2
+
+            omni = OmnibusEmbed(n_components=3, diag_aug=diag_aug)
+            OmniBar = compute_bar(omni.fit_transform([csr_matrix(A1), csr_matrix(A2)]))
+
+            omni = OmnibusEmbed(n_components=3, diag_aug=diag_aug)
+            ABar = compute_bar(omni.fit_transform([Abar, Abar]))
+
+            tol = 1.0e-2
+            np.testing.assert_allclose(
+                norm(OmniBar, axis=1), norm(ABar, axis=1), rtol=tol, atol=tol
```

Comment:
should `ABar` here be sparse or does it not matter?

Reply:
The return type of omni will still be ndarrays -- the only thing that is made sparse is the omnibus matrix which is used to calculate the SVD. `latent_left`/`latent_right` will have to be dense